### PR TITLE
periph_common/I2C: introduce new return codes scheme

### PR DESCRIPTION
**Pull request by dylad**
### Contribution description
This PR changes the API return codes in order to use errno.

I don't master doxygen mysteries so If there is better way don't hesitate to tell me.

### Issues/PRs references
#6577

### CODE REVIEW

File: drivers/periph_common/i2c.c [hunk 1]
```diff
@@ -24,23 +24,22 @@
 
 #ifdef I2C_NUMOF
 
-#ifdef PERIPH_I2C_NEED_READ_REG
+#ifdef PERIPH_I2C_NEED_READ_REGS
```
Comment (aabadie) [inline drivers/periph_common/i2c.c]: Seems unrelated ?

File: drivers/include/periph/i2c.h [hunk 2]
```diff
@@ -272,9 +240,13 @@ int i2c_release(i2c_t dev);
  * @param[out] data         memory location to store received data
  * @param[in]  flags        optional flags (see @ref i2c_flags_t)
  *
- * @return                  I2C_ACK on successful transfer of @p len byte
- * @return                  I2C_ADDR_NACK if response to address byte was NACK
- * @return                  I2C_ERR for any other error
+ * @return                  0 Success
+ * @return                  -EIO Slave device doesn't ACK the byte
```
Comment (aabadie) [inline drivers/include/periph/i2c.h]: maybe add a `when` between error code and the rest. `-EIO when slave device doesn't ACK the byte`.

File: drivers/periph_common/i2c.c (hunk 1 above)
Comment (dylad) [reply drivers/periph_common/i2c.c]: Oops, will fix

File: drivers/include/periph/i2c.h [hunk 3]
```diff
@@ -240,13 +240,13 @@ int i2c_release(i2c_t dev);
...
- * @return                  -EIO Slave device doesn't ACK the byte
- * @return                  -ENXIO No devices respond on the address sent on the bus
- * @return                  -ETIMEDOUT  Timeout occurs before device's response
- * @return                  -EINVAL Invalid argument given
- * @return                  -EOPNOTSUPP MCU driver doesn't support the flag operation
- * @return                  -EAGAIN  Lost bus arbitration
+ * @return                  0 When success
+ * @return                  -EIO When slave device doesn't ACK the byte
+ * @return                  -ENXIO When no devices respond on the address sent on the bus
+ * @return                  -ETIMEDOUT  When timeout occurs before device's response
```
Comment (aabadie) [inline drivers/include/periph/i2c.h]: There's an extra space before `When` here and after -EAGAIN also. Squash directly

**Comment by dylad** [comment]
@aabadie I've addressed your comments

**Comment by dylad** [comment]
Squashed.
I'm wondering what we can do about ret code for acquire/release ?

File: drivers/periph_common/i2c.c [hunk 4]
```diff
@@ -37,7 +37,7 @@ int i2c_read_regs(i2c_t dev, uint16_t addr, uint16_t reg,
                   void *data, size_t len, uint8_t flags)
 {
     /* First set ADDR and register with no stop */
-    int8_t ret = i2c_write_bytes(dev, addr, &reg, (flags & I2C_REG16) ? 2 : 1,
+    int ret = i2c_write_bytes(dev, addr, &reg, (flags & I2C_REG16) ? 2 : 1,
```
Comment (aabadie) [inline drivers/periph_common/i2c.c]: The alignment with the second line is incorrect now. Same below

**Comment by aabadie** [comment]
> I'm wondering what we can do about ret code for acquire/release ?

There's an EBUSY error code that would be useful with i2c_require. But I think it's fine like it is now (0 on success, -1 on error), no need for more.

File: drivers/periph_common/i2c.c (hunk 4 above)
Comment (dylad) [reply drivers/periph_common/i2c.c]: Should be good now

**Comment by aabadie** [comment]
Now it would be great if someone from the @RIOT-OS/admin team could remove Murdock from the list of required status checks in the `new_i2c_if` branch protection settings. Because in the actual status of this branch, Murdock will always fail...

**Comment by aabadie** [comment]
The green button is back, let's merge
